provider_shopper/test/cart_widget_test.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+import 'package:provider_shopper/screens/cart.dart';
+
+CartModel cartModel;
+CatalogModel catalogModel;
+Widget createCartScreen() => MultiProvider(
+      providers: [
+        Provider(create: (context) => CatalogModel()),
+        ChangeNotifierProxyProvider<CatalogModel, CartModel>(
+          create: (context) => CartModel(),
+          update: (context, catalog, cart) {
+            catalogModel = catalog;
+            cartModel = cart;
+            cart.catalog = catalogModel;
+            return cart;
+          },
+        ),
+      ],
+      child: MaterialApp(
+        home: MyCart(),
+      ),
+    );
+
+void main() {
+  group('CartScreen widget tests', () {
+    testWidgets('Testing when the cart is empty', (tester) async {
+      await tester.pumpWidget(createCartScreen());
+
+      // Testing the layout of the screen widgets
+      expect(find.text('Cart'), findsOneWidget);
+      expect(find.text('\$0'), findsOneWidget);
+      expect(
+          find.ancestor(
+              of: find.text('BUY'), matching: find.byType(FlatButton)),
+          findsOneWidget);
+      // verify no snacknar initially
```

Comment:
nit: "Verify no snackbar initially exists."

Within the repo, comments start with a capital letter and end with a period, so it'd be good to make sure all comments added in this PR do the same. It's a grammar annoyance, I admit, but those little details do matter. :)